lxd/instance/drivers/driver_qemu.go

```diff
@@ -7464,8 +7464,40 @@ func (d *qemu) CGroup() (*cgroup.CGroup, error) {
 	return nil, instance.ErrNotImplemented
 }
 
-// SetAffinity is not implemented for VMs.
+// SetAffinity sets affinity for QEMU processes according with a set provided.
 func (d *qemu) SetAffinity(set []string) {
+	monitor, err := qmp.Connect(d.monitorPath(), qemuSerialChardevName, d.getMonitorEventHandler())
+	if err != nil {
+		// this is not an error path, really. Instance can be stopped, for example.
+		d.logger.Debug("Failed connecting to the QMP monitor. Instance is not running?", logger.Ctx{"name": d.Name(), "err": err})
+		return
+	}
+
+	// Get the list of PIDs from the VM.
+	pids, err := monitor.GetCPUs()
+	if err != nil {
+		d.logger.Error("Failed to get VM instance's QEMU process list", logger.Ctx{"name": d.Name(), "err": err})
+		return
+	}
+
+	// Confirm nothing weird is going on.
+	if len(set) != len(pids) {
+		d.logger.Error("QEMU has less vCPUs than configured", logger.Ctx{"name": d.Name()})
```

Comment:
Can you include the pids and set length in the error returned to help with diagnostics?

Reply:
Fixed! Thanks!